### ISSUE TITLE
List of Potential Features

### ISSUE THREAD

**Issue by SilentChaos512** [SilentChaos512/Silents-Mechanisms#8]
Just a list of some general goals for the mod. Any item on the list may or may not become a feature.

**Please limit comments to just feature suggestions.** Report bugs in a new issue. For questions, use [my Discord server](https://discord.gg/gh84eWK) (#silents-mechanisms channel, please do not DM).

- [x] Machine progression
    - Basic machines are cheaper to craft, but slower and do not support upgrades
    - Standard machines require alloys to craft, work faster, and support upgrades
- [x] Machine upgrades
    - [x] Increase speed
    - [x] Increase output chance
    - [ ] Energy capacity/efficiency
- [x] Ability for machines to respond to redstone signals (disable a generator with redstone, etc.)
- [x] ComputerCraft compatibility
    - Read energy levels
    - Set redstone mode
- [ ] More generators
    - [x] Lava
    - [ ] Solar
    - [ ] Nuclear reactor?
- [x] Wires
    - [ ] Remember connection type
    - [x] ~Maybe change default type to output?~ Smarter default connection type
- [x] Stackable fluid container items (canisters?)
- [ ] Fluid pipes
- [x] Oil
    - [x] Refinery (with "recipe" system)
    - [x] Diesel
    - [x] Plastics
- [x] Mixer (fluid machine with multiple inputs, one output)
- [x] Pump
- [ ] Hand pump for picking up fluids from the world and draining machines (fills canisters, maybe buckets?)
- [ ] Melter/Liquifier (items to fluids)
- [ ] Bottler (fills buckets and canisters)
- [ ] Fluid tank
    - [ ] Fluid tank mine cart
- [ ] Automatic crafting table
- [ ] More metal crafting items
    - [ ] Gears
    - [ ] Plates
    - [ ] Rods
    - [ ] Dense plates maybe?
- [ ] Mining machine (like a quarry, etc.)

**Comment by ghost** [comment]
does this mod work with Universal Cables?

**Comment by SilentChaos512** [comment]
@hackaholic5069 If they can transfer FE, it should.

**Comment by Willy347** [comment]
will you consider standard machine recipes that use the simple version so you don't end up with useless machines once you upgrade ?

**Comment by revdave33** [comment]
An oil/diesel/ethane/etc generator?

**Comment by ChilledDayZ07** [comment]
I have a suggestion for a new machine. I know you have a lot to do already but thought I would see what you think. As you have a solidifier added maybe you could add a melter for ore tripling and maybe alloying in the mixer and put thought the solidifier to make ingots.

**Comment by ChilledDayZ07** [comment]
I would like to suggest adding a bio diesel generator in the future, it wouldn’t make as much FE as the diesel gen but would be a more renewable source of power?

**Comment by RamenNoodles** [comment]
Using MC1.14.4
I've been trying to get mods to store liquid the pump extracts but nothing seems to work from whatever fluid pipe or tank mods for mc1.14.4.  I'm guessing the pump just isn't pushing anything from it's inventory except for the bucket or canister.  This is fine but at least allow the canister to interact with other mods. I guess it would need to work like a filled bucket but stackable and I'm guessing that isn't possible?

**Comment by SilentChaos512** [comment]
@RamenNoodles You should open a new issue, please stay on topic when commenting on other issues. I do realize the fluid canisters can't work with other mods. I'm not aware of any way to fix that at the moment.

(edit) I'll go ahead and make the pump issue

**Comment by RamenNoodles** [comment]
@SilentChaos512 Sorry about that. I didn't think of it as an issue since you had fluid tanks and pipes in the pipeline.  My issue would have to be something I would have to deal with until then.  If I were to make an issue it would have to be with the canisters, I'll probably go ahead and do that.

**Comment by McKitten** [comment]
Now that Vazkii has released Botania for 1.14, overworld production of blaze powder and glowstone are available to users without entering the nether. Whilst I'd prefer the ability to move liquid between machines be given priority, some compression recipes for the blaze powder would be appreciated as would crusher recipes for standard vanilla items like sandstone & clay.

**Comment by SilentChaos512** [comment]
@McKitten What recipes specifically should be added? Recipes are really easy, takes less than a minute to add each one.

**Comment by McKitten** [comment]
In terms of crusher stuff, Sandstone back to sand/red sand, clay back into balls, gravel into flint (probably needed the most due to arrows). Maybe even wheat into seeds and leather saddles into leather.
In terms of compression i'd like to be able to reform blaze rods, a lossy recipe is still better then none.

**Comment by SilentChaos512** [comment]
Gravel already crushes into sand. I'll consider the others. Don't forget
that you can always add recipes with data packs if they aren't included
with any mod.

**Comment by McKitten** [comment]
This is true assuming models exist within minecraft already; an issue i once discovered by trying to add more compression items to its mod. I thought it wise to double check.

**Comment by Wabbit0101** [comment]
Enhancement (or new machine): slurry mixer: an extension of mixer than can take non-fluids as ingredients (still limited to total of 4 items). The output is a single fluid. 

Example uses: concrete or cement, compost tea (fertilizers), (food) syrups, slurries (eg food production), sewage, industrial-scale potions, ciders, paints, etc.

**Comment by CameoDaManeo** [comment]
Hey. I would like to suggest making bismuth steel compatible as a forge:steel_ingot. ATM, bismuth has hardly any uses and you end up with a tonne of it in a playthrough. Bismuth steel ingots are cheaper to make than steel ingots, so making them compatible with forge:steel_ingot would motivate people to actually make bismuth steel.

In the same respect, could you also do this with bismuth brass and create other bisnuth alloys that don't currently exist, such as bismuth bronze?

**Comment by TheNimbleNinja** [comment]
can you add a white list feature to the mixer as it keeps getting full of fluid that shouldnt go in it
